spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -380,15 +452,20 @@ private File handleFileMessage(final File sourceFile, File tempFile, final File
 
 	private File handleInputStreamMessage(final InputStream sourceFileInputStream, File originalFile, File tempFile,
 										  final File resultFile) throws IOException {
-		if (FileExistsMode.APPEND.equals(this.fileExistsMode)) {
-			File fileToWriteTo = this.determineFileToWrite(resultFile, tempFile);
-			final BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(fileToWriteTo, true));
+		final boolean append = FileExistsMode.APPEND.equals(this.fileExistsMode)
+				|| FileExistsMode.APPEND_NO_FLUSH.equals(this.fileExistsMode);
+
+		if (append) {
+			final File fileToWriteTo = this.determineFileToWrite(resultFile, tempFile);
+
+			final FileState state = getFileState(fileToWriteTo, false);
 
 			WhileLockedProcessor whileLockedProcessor = new WhileLockedProcessor(this.lockRegistry,
 					fileToWriteTo.getAbsolutePath()) {
 
 				@Override
 				protected void whileLocked() throws IOException {
+					BufferedOutputStream bos = state != null ? state.stream : createOutputStream(fileToWriteTo, append);
```

Comment:
IDEA says me that this `append` argument is always `true`.
And it is according to the `if (append) {` logic.
Minor, of course, but I know how you don't like the IDE noise :smile: